test/test_images.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter_test/flutter_test.dart';
+
+
+class FakeHttpClient extends Fake implements HttpClient {
```

Comment:
```
test [nfc]: Move fake-image-request code to its own file
```

N.B.: This now has the same name as `FakeHttpClient` in test/api/fake_api.dart, which causes awkwardness when a file wants import lines for both files. Later in this branch, this awkwardness shows up like this:

```diff
-import '../api/fake_api.dart';
+import '../api/fake_api.dart' as fake_api;
```

Maybe there's a better name for this class, like `FakeImageHttpClient`? (Then maybe also its friends, like `FakeHttpClientResponse`, get renamed correspondingly?)